core/src/main/java/org/infinispan/container/offheap/OffHeapDataContainer.java

```diff
@@ -494,9 +491,13 @@ public void evict(WrappedBytes key) {
       lock.lock();
       try {
          checkDeallocation();
-         // TODO: this could be more efficient
-         passivator.passivate(get(key));
-         remove(key);
+         long address = memoryLookup.getMemoryAddress(key);
+         if (address != 0) {
+            // TODO: this could be more efficient
+            InternalCacheEntry<WrappedBytes, WrappedBytes> ice = performGet(address, key);
+            passivator.passivate(ice);
```

Comment:
We should add an if check here for ice and only do this line and the next if it is not null. Prevents another possible loop through the linked list if it exists.

Reply:
OK, I was only inlining the calls, but I can add it.